pkg/plugins/golang/v4/scaffolds/internal/templates/controllers/controller.go

```diff
@@ -85,9 +85,9 @@ type {{ .Resource.Kind }}Reconciler struct {
 	Scheme *runtime.Scheme
 }
 
-// +kubebuilder:rbac:groups={{ .Resource.QualifiedGroup }},resources={{ .Resource.Plural }},verbs=get;list;watch;create;update;patch;delete
-// +kubebuilder:rbac:groups={{ .Resource.QualifiedGroup }},resources={{ .Resource.Plural }}/status,verbs=get;update;patch
-// +kubebuilder:rbac:groups={{ .Resource.QualifiedGroup }},resources={{ .Resource.Plural }}/finalizers,verbs=update
+// +kubebuilder:rbac:groups={{ if .Resource.External }}{{ .Resource.Domain }}{{ else }}{{ .Resource.QualifiedGroup }}{{ end }},resources={{ .Resource.Plural }},verbs=get;list;watch;create;update;patch;delete
```

Comment:
I think its fine as it was before using the QualifiedGroup here.

The thing with certmanager (and potentially other external APIs) that they do not necessarily follow the `{domain}/{group}` schema that is used by kubebuilder. The full qualified group name for certmanager is simple `cert-manager.io` without a "sub-group". 

I think when scaffoling a controller for this you should probably have a empty `--group`. Currently the group is set via `--group certmanager`. Would that solve the issues you face at the moment?

Reply:
The QualifiedGroup group seems to make sense for Kubebuilder APIs. 
We should not really allow the group be empty since it will result in a scaffold that will not work if more than one API be scaffolded.

Reply:
Hi @fischor,

Thank you so much! That’s exactly the help I was looking for. We discussed the scenarios to ensure that it will work well.

Your comment made me think: actually, CertManager does not define domains for their APIs, just the group. So, we must accommodate both scenarios—allowing the API to be scaffolded with only the group or with both the group and the domain.

As properly documented, we will ensure that both scenarios are tested with appropriate testdata generation.
The changes were applied on this PR.

Reply:
Hi @fischor,

Thank you so much! That’s exactly the help I was looking for. We discussed the scenarios to ensure that it will work well.

Your comment made me think: actually, CertManager does not define domains for their APIs, just the group. So, we must accommodate both scenarios—allowing the API to be scaffolded with only the group or with both the group and the domain.   In this way, see that we are fixing here how we scaffold and allowing the scaffold be done with an empty domain.